packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx

```diff
@@ -122,12 +122,13 @@ const GatsbyImageHydrator: FC<GatsbyImageProps> = function GatsbyImageHydrator({
           ssrImage.removeAttribute(`data-gatsby-image-ssr`)
         }, 0)
       } else {
-        document.addEventListener(`load`, function onLoadListener() {
-          document.removeEventListener(`load`, onLoadListener)
+        onStartLoad?.({
+          wasCached: true,
+        })
+
+        ssrImage.addEventListener(`load`, function onLoadListener() {
```

Comment:
I'm assuming we used `document` here because `ssrImage` would be falsy in this else block. Was this change necessary to make things work? If yes, we should probably check for truthy `ssrImage` before using it

Reply:
I think you might be looking at the outer `if` condition, this `else` corresponds to `if (ssrImage.complete) { ... }`. 

That was the main change to make it work, I'm not quite sure why it was `document` to begin with either (if we listen on `document` we would need to inspect the event target to differentiate for it to work that way anyway)

Reply:
oh, lol, yeah you're right. Missed that line 111